compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -947,7 +949,7 @@ impl<'tcx> LayoutOfHelpers<'tcx> for CodegenCx<'_, 'tcx> {
     #[inline]
     fn handle_layout_err(&self, err: LayoutError<'tcx>, span: Span, ty: Ty<'tcx>) -> ! {
         if let LayoutError::SizeOverflow(_) = err {
-            self.sess().span_fatal(span, &err.to_string())
+            self.sess().emit_fatal(LayoutSizeOverflow { span, error: err.to_string() })
```

Comment:
`LayoutError` implements `IntoDiagnostic`, can it be used directly here? I vaguely recall suggesting we implement `IntoDiagnostic` for `Spanned<T>` where `T: IntoDiagnostic` so that a non-spanned type like `LayoutError` can be emitted with a span without introducing a distinct type combining the two.

Reply:
I've had no idea Spanned existed, I can use it, sure. However with 
```rs
    #[inline]
    fn handle_layout_err(&self, err: LayoutError<'tcx>, span: Span, ty: Ty<'tcx>) -> ! {
        if let LayoutError::SizeOverflow(_) = err {
            self.sess().emit_fatal(Spanned { span, node: err })
        } else {
            span_bug!(span, "failed to get layout for `{}`: {}", ty, err)
        }
    }      
```
I have to ensure that both `'a` and `'ll` lifetimes from `&'a self` and `CodegenCx<'ll, 'tcx>` must outlive `'tcx`. And I'm unsure if I can deal with it without changing the trait's implementation.

Reply:
Maybe we can just use a wrapper `SpannedLayoutError` for now, which manually implements `IntoDiagnostic, and just calls `into_diagnostic` on the layout error field, and then calls `set_span` with the `Span` field?

Reply:
Using this wrapper would lead to the same problem as when using `Spanned`. Instead, I've changed implementation of `IntoDiagnostic` for `LayoutError` to not bind its lifetime to the `Handler`'s and returned `DiagnosticBuilder`'s lifetime. This resolved the problem of lifetimes, so I was able to use `Spanned` as you've suggested previously.